public/coffee/ide/editor/directives/aceEditor/spell-check/SpellCheckManager.coffee

```diff
@@ -246,3 +250,28 @@ define [
 				error: (xhr, status, error) ->
 					callback error
 			return $.ajax options
+
+		blacklistedCommandRegex: ///
+			\\                    # initial backslash
+			(label                # any of these commands
+			|[a-z]{0,2}ref
+			|usepackage
+			|begin
+			|end
+			|[a-z]{0,2}cite
+			|input
+			|include
+			|includegraphics)
+			( \[ [^\]]* \] )?     # optional [...] args
+				\{ [^}]*  \}        # the {...} args
+		///g
+
+		blankOutBlacklistedCommands: (line) ->
+			line.replace @blacklistedCommandRegex, (command) ->
+				command.replace(
```

Comment:
Why do we need to do any analysis of the match, can we not just blank the whole thing? Am I right in saying this currently returns, e.g. `\usepackage[.........]{.............}`. Can we just return all `.............................` instead?